src/drivers/fs/virtio_fs.rs

```diff
@@ -195,9 +203,38 @@ impl FuseInterface for VirtioFsDriver {
 		let mut transfer_result =
 			self.vqueues[1].dispatch_blocking(buffer_tkn, BufferType::Direct)?;
 
-		let headers = unsafe { transfer_result.used_recv_buff.pop_front_downcast().unwrap() };
-		let payload = transfer_result.used_recv_buff.pop_front_vec();
-		Ok(Rsp { headers, payload })
+		let (dyn_headers, written_header_len) =
+			transfer_result.used_recv_buff.pop_front_raw().unwrap();
+		let headers = dyn_headers
+			.downcast::<MaybeUninit<RspHeader<O, MaybeUninit<O::OutStruct>>>>()
+			.unwrap();
+		if written_header_len < size_of::<fuse_out_header>() {
+			Err(VirtqError::IncompleteWrite.into())
+		} else {
```

Comment:
Can we do early returns instead of this huge expression with increasing indentation (here and further down)?

```suggestion
		if written_header_len < size_of::<fuse_out_header>() {
			return Err(VirtqError::IncompleteWrite.into());
		}
```